bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -483,14 +486,32 @@ public boolean isEmpty(SerializerProvider provider, String value) {
      * @return {@code Map<String, String>} returns all custom property key value pairs associated with the resource
      */
     private Map<String, String> getCustomProperties() {
+        Map<String, String> customProperties = new HashMap<>();
+        Map<String, String> templateBasedCustomProperties;
+        List<String> excludedPrefixes = Arrays.asList("fd:", "jcr:", "sling:");
```

Comment:
Please add test cases in exporter for these prefixes. What if there is a property which starts with `fd:` and I want it to be a custom property. I guess the below code won't add this in custom properties.

Will individual classes override it ?

Reply:
By default any property with the `fd:` prefix will have to be handled separately by each class. Just like we have done for `fd:viewType` [here](https://github.com/adobe/aem-core-forms-components/blob/4f37812634959bb0d4e7651add6a47ff287328d3/bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java#L127)

Reply:
or, for common props, we could handle it just like it's handled for fd:dor. (after `getCustomProperties` method is invoked in `getProperties`)

Reply:
> or, for common props, we could handle it just like it's handled for fd:dor. (after `getCustomProperties` method is invoked in `getProperties`)

Yes, if the property is common for all field, it could be done this way too.

Reply:
Added `testGetCustomProperties` in TextInputImplTest